codeframe/ui/server.py

```
@@ -583,6 +585,139 @@ async def get_chat_history(project_id: int, limit: int = 100, offset: int = 0):
+            db=app.state.db,
+            api_key=api_key
+        )
+
+        # CRITICAL: Validate discovery is active before processing answer
+        status = agent.get_discovery_status()
+        if status.get('state') != 'discovering':
+            raise HTTPException(
+                status_code=400,
+                detail=f"Discovery is not active. Current state: {status.get('state')}. "
+                       f"Please start discovery first by calling POST /api/projects/{project_id}/discovery/start"
+            )
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)